src/scripts/main.js

```diff
@@ -1,3 +1,30 @@
 'use strict';
 
-// write code here
+const ul = document.querySelector('ul');
+const list = [...ul.querySelectorAll('li')];
+
+function toNumber(str) {
+  return +str.replace(/['$']/g, '').replace(/[',']/g, '.');
+}
+
+function sortList() {
+  list.sort(
+    (li1, li2) => toNumber(li2.dataset.salary) - toNumber(li1.dataset.salary),
+  );
+
+  for (const li of list) {
+    ul.append(li);
```

Comment:
Appending elements to the DOM in a loop can cause performance issues, especially with large lists. Consider using a DocumentFragment to minimize reflows and improve performance.